tests/utils.py

```diff
@@ -123,7 +123,11 @@ def apply(self, **kwargs):
         t_end = kwargs['t_end']
         assert(t_start <= t_end)
         if self.direction == 1:
-            self.u[:] = self.u[:] + self.direction * abs(t_start - t_end)
+            for t in range(t_start, t_end):
+                idx = t % np.shape(self.u)[0]
+                past_idx = (t-1) % np.shape(self.u)[0]
```

Comment:
You're assuming that the operator moves one time step at a time, when this has is supposed to go from `t_start` to `t_end`.